### ISSUE TITLE
Support Flatcar Linux on Hetzner Cloud

### ISSUE THREAD

**Issue by P4sca1** [kubermatic/kubeone#3249]
As a Hetzner Cloud user
I would like to deploy worker nodes using the Flatcar Linux operating system. I already run my control plane using Flatcar on Hetzner Cloud (both ARM and x86 successfully.

Official Flatcar docs for Hetzner: https://www.flatcar.org/docs/latest/installing/cloud/hetzner/

Hetzner support is currently only available on the beta channel, and not in the stable channel.

Hetzner cloud does not provide a default image for Flatcar, however it is possible to create a snapshot for Flatcar Linux manually, which can be used as an image when creating new servers.

For initial beta support, it would be great if kubeone allows the operatingSystem: flatcar annotation for Hetzner worker nodes, as long as a custom image is provided. The user would be responsible for creating the Flatcar Linux snapshot in Hetzner Cloud.

**Comment by P4sca1** [comment]
Maybe the machine-controller or operating-system-manager repository is a better fit for this feature request. In that case, feel free to transfer this issue to another project.

**Comment by kron4eg** [comment]
@P4sca1 is there something in the code that prevents this to happen?

**Comment by P4sca1** [comment]
@kron4eg Yes, I think it it because of the machine-controller validation logic:

https://github.com/kubermatic/machine-controller/blob/main/pkg/cloudprovider/provider/hetzner/provider.go#L81
